src/component/requests.js

```diff
@@ -15,32 +17,69 @@ export default class Requests extends React.Component {
     });
   }
 
+  componentDidMount() {
+    const requests = React.findDOMNode(this);
+    requests.addEventListener('scroll', () => {
+      const fromIndex = Math.ceil(requests.scrollTop / requestElementHeight) - 15;
+      this._setFromIndex(fromIndex < 0 ? 0 : fromIndex);
+    });
+  }
+
+  componentDidUpdate() {
+    const previousFilter = this.filter;
+    this.filter = this.props.requestData.filter;
+    const requests = React.findDOMNode(this);
```

Comment:
Considering `componentDidUpdate` is called on every "scroll-frame", is `React.findDOMNode` a fast operation? Otherwise I would put it into the if condition (where you really need it).